src/runtime_src/core/common/ishim.h

```diff
@@ -271,6 +274,13 @@ struct shim : public DeviceType
       throw error(ret, "failed to get BO properties");
   }
 
+  virtual void
+  get_device_info(struct xclDeviceInfo2 *info) const
+  {
+    if (auto ret = xclGetDeviceInfo2(DeviceType::get_device_handle(), info))
+      throw error(ret, "failed to get device info");
+  }
+
```

Comment:
Not needed, device_linux is part of libxrt_core and can call xclGetDeviceInfo2 directly.

Reply:
removed